sway-core/src/semantic_analysis/ast_node/expression/typed_expression/struct_instantiation.rs

```diff
@@ -21,13 +21,18 @@ pub(crate) fn struct_instantiation(
     let decl_engine = ctx.decl_engine;
     let engines = ctx.engines();
 
+    // Populate call_path_binding with real types
```

Comment:
Can you add a longer comment here explaining what this block of code is doing? LOC 25 - 74? In regards to this being for the LSP, the `type_binding` object being used for the LSP, etc.

In fact, I'm pleasantly surprised that LOC 25 isn't conflicting with the hack on LOC 36 for resolving the self type. LOC 36 was added because the `Self` keyword was unable to be parsed in this position (at some point in the past).